python/tank/descriptor/descriptor.py

```diff
@@ -371,6 +376,40 @@ def find_latest_version(self, constraint_pattern=None):
         latest._io_descriptor = self._io_descriptor.get_latest_version(constraint_pattern)
         return latest
 
+    def find_latest_cached_version(self, constraint_pattern=None):
```

Comment:
My initial reaction was that maybe this should be an additional option to the `find_latest_version` method. Wondering if you'd thought about doing it that way.